Source/Dafny/Compilers/Compiler-js.cs

```diff
@@ -2401,6 +2402,14 @@ protected class ClassWriter : IClassWriter {
         }
         nodeProcess.StandardInput.Flush();
         nodeProcess.StandardInput.Close();
+        int current;
+        // Fixes a problem of Node on Windows, where Node does not prints to the parent console its standard outputs.
+        while ((current = nodeProcess.StandardOutput.Read()) != -1) {
+          Console.Out.Write((char)current);
+        }
+        while ((current = nodeProcess.StandardError.Read()) != -1) {
+          Console.Error.Write((char)current);
+        }
```

Comment:
I don't think this is safe, the process may fill the stderr buffer and block on it while we wait for stdout.

Reply:
From https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.standardoutput?view=net-6.0:

> Synchronous read operations introduce a dependency between the caller reading from the [StandardOutput](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.standardoutput?view=net-6.0) stream and the child process writing to that stream. These dependencies can result in deadlock conditions. When the caller reads from the redirected stream of a child process, it is dependent on the child. The caller waits on the read operation until the child writes to the stream or closes the stream. When the child process writes enough data to fill its redirected stream, it is dependent on the parent. The child process waits on the next write operation until the parent reads from the full stream or closes the stream. The deadlock condition results when the caller and child process wait on each other to complete an operation, and neither can proceed. You can avoid deadlocks by evaluating dependencies between the caller and child process.

Reply:
I pushed something so that the error writing is done in a different thread. I think it solves the deadlock problem. Let me know if you agree with this.